bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/zk/ZKMetadataDriverBase.java

```diff
@@ -260,6 +262,47 @@ public synchronized LedgerManagerFactory getLedgerManagerFactory()
         return lmFactory;
     }
 
+    public CompletableFuture<Void> disableHealthCheck() {
+        CompletableFuture<Void> createResult = new CompletableFuture<>();
+        try {
+            zk.create(getEnableHealthPath(), new byte[0], ZkUtils.getACLs(conf), CreateMode.PERSISTENT);
+            createResult.complete(null);
+        } catch (Exception e) {
+            createResult.completeExceptionally(e);
+        }
+        return createResult;
+    }
+
+    public CompletableFuture<Void>  enableHealthCheck() {
+        CompletableFuture<Void> deleteResult = new CompletableFuture<>();
+
+        try {
+            zk.delete(getEnableHealthPath(), -1);
+            deleteResult.complete(null);
+        } catch (Exception e) {
+            deleteResult.completeExceptionally(e);
+        }
+        return deleteResult;
+    }
+
+    public CompletableFuture<Boolean> isEnableHealthCheck() {
+        CompletableFuture<Boolean> enableResult = new CompletableFuture<>();
+        try {
+            boolean isEnable = (null == zk.exists(getEnableHealthPath(), false));
+            enableResult.complete(isEnable);
+        } catch (Exception e) {
+            enableResult.completeExceptionally(e);
+        }
+        return enableResult;
+    }
+
+
+    public String getEnableHealthPath() {
+        String zkLedgersRootPath = ZKMetadataDriverBase.resolveZkLedgersRootPath(conf);
```

Comment:
we should cache this return value in the `initialize` method. we don't need to recalculate it everytime

Reply:
OK ,I will fixed

Reply:
The same directory seems to be defined, can I use ledgerRootPath directly?
https://github.com/apache/bookkeeper/blob/370d785ff674841943fa8cddee1964408e826f5b/bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/zk/ZKMetadataDriverBase.java#L143~L145

Reply:
I meant the whole zk path, including /enableHealthCheck, so you don't need computation in the "hot" path

Reply:
OK, I will fixed